### ISSUE TITLE
[Bug] - Docker target channel issue

### ISSUE THREAD

**Issue by Slopz** [jenslys/autovod#55]
Docker container keeps setting twitch.tv/dockeruser as the target channel

```
18:29:54| Selected streamer: dockeruser
18:29:54| Found config file
18:29:54| Starting AutoVOD
18:29:54| Loading dockeruser.config
18:29:54| Checking twitch.tv/dockeruser for a stream
18:29:54| All required files found
error: No playable streams found on this URL: twitch.tv/dockeruser
18:29:58| Trying again in 1 minute
```

cp default.config targetchannel.config
sudo docker build --build-arg TWITCH_USER=targetchannel -t autovod .
sudo docker run -d autovod

despite the channel being live, it's not picking it up.
am I doing something wrong?

autovod v2.0.0

**Comment by jenslys** [comment]
The dockerfile was created by @BrainyXS, as I currently don't have much knowledge on docker. So I will leave that up to him. 

cc: @BrainyXS

**Comment by totallyparity** [comment]
Long comment coming up; Apologies in advance.

With some trial and error, by changing few lines from `AutoVOD.sh` and `DockerFile` I was able to get `AutoVOD.sh` check the channel properly inside the container. I have no clue if those few changes could break the script however.
```DockerFile
# DockerFile
FROM alpine # This will use 'alpine:latest' image

# Define an ARG variable & set an ENV variable for AutoVOD.sh later

ARG	TWITCH_USER # ARG is used only when building; No effect on container itself. (--build-arg)
ENV	NAME=${TWITCH_USER} # This Environment variable will be set for built container.

# Upgrade the system and install dependencies

RUN	apk add --no-cache --upgrade python3 tar wget bash jq rclone curl # Added curl for curl command in the shell script
RUN	python3 -m ensurepip

RUN	pip3 install --upgrade streamlink 
RUN	wget wget https://github.com/porjo/youtubeuploader/releases/download/23.02/youtubeuploader_23.02_Linux_arm64.tar.gz
RUN	tar -xvf youtubeuploader_23.02_Linux_arm64.tar.gz && rm youtubeuploader_23.02_Linux_arm64.tar.gz &&\
	mv youtubeuploader /usr/local/bin/youtubeuploader
# Maybe steps above could be written better with architecture check routine?


# Copy the required files

COPY	${TWITCH_USER}.config /autoVOD/${TWITCH_USER}.config # Changed to fit $NAME variable for later
COPY	AutoVOD.sh /autoVOD/AutoVOD.sh
COPY	client_secrets.json /autoVOD/client_secrets.json
COPY	request.token /autoVOD/request.token


# Start AutoVod

WORKDIR /autoVOD
CMD	["bash", "AutoVOD.sh"]
```
`--build-arg` for `docker build` wouldn't work for AutoVOD Docker container as `--build-arg` is only used when building an image for container. And then there's this problem:
https://github.com/jenslys/autovod/blob/3bbac74288a355b19d459077726b4ea6ef8dfe36/Dockerfile#L27
Looking at this line, it seems the issue @Slopz mentioned is caused from this. Since script tries to fetch arguments from shell command you would enter manually, it will utilize '-ndockeruser' from command and it will try to check 'twitch.tv/dockeruser'. In order to fix this I added an environment variable `NAME` which is also used in `AutoVOD.sh`.
```DockerFile
ARG	TWITCH_USER
ENV	NAME=${TWITCH_USER}
```
With that, I will change the shell command from `AutoVOD.sh` like so:
```DockerFile
CMD	["bash", "AutoVOD.sh"]
```
Since `-n` isn't necessary, we don't need to use `fetchArgs` function either:
https://github.com/jenslys/autovod/blob/3bbac74288a355b19d459077726b4ea6ef8dfe36/AutoVOD.sh#L7-L19
From `AutoVOD.sh`, we could just comment `fetchArgs` function out so that `NAME` would remain unchanged since configured from `DockerFile`:
```bash
#fetchArgs "$@"
```
Maybe just clearing out the function for later would work, but then manual installation method would get little bit complicated since using Environment Variable means something like `export NAME=targetchannel` is necessary. (Maybe make a separate script only for Docker?)
https://github.com/jenslys/autovod/blob/3bbac74288a355b19d459077726b4ea6ef8dfe36/AutoVOD.sh#L52
https://github.com/jenslys/autovod/blob/3bbac74288a355b19d459077726b4ea6ef8dfe36/Dockerfile#L8
Since `AutoVOD.sh` utilizes `curl`, I also changed this line from `DockerFile` so that `curl` command works inside the container:
```DockerFile
RUN	apk add --no-cache --upgrade python3 tar wget bash jq rclone curl
```

This was the result after change, checked from container logs: (Running on Docker installed on Raspberry Pi 3B; `ENV` `NAME` is set to 'targetto' as an example)
```
17:04:23| Selected streamer: targetto
17:04:24| Found config file
17:04:24| Starting AutoVOD
17:04:24| Loading targetto.config
17:04:24| Trying to fetch stream metadata
17:04:26| Stream is online!
17:04:26| Current Title: null
17:04:26| Current Game: null
17:04:26| Checking twitch.tv/targetto for a stream
17:04:26| All required files found
error: No playable streams found on this URL: twitch.tv/targetto
17:04:34| Trying again in 1 minute
```
 There could be a better way than what I did, but that's all from me.

**Comment by jenslys** [comment]
> 

Hi,
I appreciate the debugging, do you mind compiling this into a PR? so it's easier for me to see the suggested changes and test it out.

Thanks 😄

**Comment by jenslys** [comment]
Update: Seem to have a *hacky* fix on #61
